mlx/traceability.py

```diff
@@ -876,6 +955,147 @@ def process_item_nodes(app, doctree, fromdocname):
+
+                            if source_id not in linked_attributes.keys():
+                                linked_attributes[source_id] = attribute_value
+                            else:
+                                # store newly encountered attribute value if it has a higher priority
+                                stored_attribute_priority = priorities[linked_attributes[source_id]]
+                                latest_attribute_priority = priorities[attribute_value]
+                                if latest_attribute_priority > stored_attribute_priority:
+                                    linked_attributes[source_id] = attribute_value
+
+        # initialize dictionary and increment counters for each possible value
+        all_states = {}
+        for priority in priorities.keys():
+            all_states[priority] = 0
+        for attribute in linked_attributes.values():
+            all_states[attribute] += 1
+
+        count_total = len(linked_attributes)
+        count_uncovered = all_states[node['label_set'][0]]
+        count_covered = count_total - count_uncovered
+        try:
+            percentage = int(100 * count_covered / count_total)
+        except ZeroDivisionError:
+            percentage = 0
+        disp = 'Statistics: {cover} out of {total} covered: {pct}%'.format(cover=count_covered,
+                                                                           total=count_total,
+                                                                           pct=percentage,)
+
+        # remove items with count value equal to 0
+        all_states = {k: v for k, v in all_states.items() if v}
+        # keep case-sensitivity of :<<attribute>>: arguments in labels of pie chart
+        case_sensitive_priorities = node['priorities']
+        for priority in case_sensitive_priorities:
+            if priority.lower() in all_states.keys():
+                value = all_states.pop(priority.lower())
+                all_states[priority] = value
+        labels = list(all_states.keys())
+
+        sizes = all_states.values()
+        explode = [0.05]  # slichtly detaches slice of first state, default is "uncovered"
+        explode.extend([0] * (len(all_states.values()) - 1))
+
+        def pct_wrapper(sizes):
+            def make_pct(pct):
+                absolute = int(round(pct / 100 * sum(sizes)))
+                return "{:.0f}%\n({:d})".format(pct, absolute)
+            return make_pct
+
+        fig, axes = plt.subplots()
+        axes.pie(sizes, explode=explode, labels=labels, autopct=pct_wrapper(sizes), startangle=90)
+        axes.axis('equal')
+        folder_name = path.join(env.app.srcdir, '_images')
+        if not path.exists(folder_name):
+            mkdir(folder_name)
+        hash_string = ''
+        for pie_slice in axes.__dict__['texts']:
+            hash_string += str(pie_slice)
+        hash_value = sha256(hash_string.encode()).hexdigest()  # create hash value based on chart parameters
```

Comment:
Can we maybe check if the image exists and then trigger at least a warning to user? I have a feeling this will bite us when document will have many pie-charts.

Reply:
I don't understand why this is a concern. Can you please elaborate?

Reply:
As discussed its not needed here - my bad understanding.